OpenRA.Game/Actor.cs

```diff
@@ -202,6 +222,11 @@ public IEnumerable<T> TraitsImplementing<T>()
 			return World.TraitDict.WithInterface<T>(this);
 		}
 
+		public T TraitInfo<T>()
```

Comment:
This is really one of the problems of the code: dependencies. Ideally, good code should have minimal understanding of the rest of the code. This is usually achieved by using factory design pattern and injecting necessary dependencies through constructor.

In order to retrieve that info object, we need to go through several classes to arrive to that information (i.e. world.worldactor.info.whatever.etc) when usually it's not necessary. This case is a clear violation of the demeter law, and this makes unit testing hellishly complicated as it will be noted in my refactor of Move activity and Mobile trait.

That function I created allowed me to make the Pathfinder trait agnostic of the actorInfo class, reducing coupling. Personally I would leave it as it is right now, and then in a future PR create a factory that constructs Pathfinder traits properly and injects dependencies as needed.

If you really wish it I could turn it back to original, just wanted to make my point.